File: dev-superior-algoritmo-e-logica-de-programacao/cpp/exercicios/acima_diagonal.cpp
#include <iostream>

using namespace std;

int main() {
    int N, soma;

    cout << "Qual a ordem da matriz? ";
    cin >> N;

    int mat[N][N];

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> mat[i][j];
        }
    }

    cout << endl << "MATRIZ DIGITADA:" << endl;
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            cout << mat[i][j] << " ";
        }
        cout << endl;
    }

    soma = 0;
    for (int i = 0; i < N; i++) {
        for (int j = i + 1; j < N; j++) {
            soma = soma + mat[i][j];
        }
    }

    cout << endl << "SOMA DOS ELEMENTOS ACIMA DA DIAGONAL PRINCIPAL = " << soma << endl;

    return 0;
}

File: dev-superior-algoritmo-e-logica-de-programacao/java/src/exercicios/AcimaDiagonal.java
package exercicios;

import java.util.Locale;
import java.util.Scanner;

public class AcimaDiagonal {

	public static void main(String[] args) {

		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);

		int N, soma;

	    System.out.print("Qual a ordem da matriz? ");
		N = sc.nextInt();

	    int[][] mat = new int[N][N];

		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
	        	System.out.printf("Elemento [%d,%d]: ", i, j);
				mat[i][j] = sc.nextInt();
	        }
	    }

		System.out.println("\nMATRIZ DIGITADA:");
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				System.out.print(mat[i][j] + " ");
			}
			System.out.println();
		}

		soma = 0;
		for (int i = 0; i < N; i++) {
			for (int j = i + 1; j < N; j++) {
				soma = soma + mat[i][j];
			}
		}

	    System.out.printf("\nSOMA DOS ELEMENTOS ACIMA DA DIAGONAL PRINCIPAL = %d\n", soma);

		sc.close();
	}
}
